jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBufferPool.java

```diff
@@ -55,4 +60,56 @@ static RetainableByteBufferPool findOrAdapt(Container container, ByteBufferPool
         }
         return retainableByteBufferPool;
     }
+
+    /**
+     * A Pool wrapper that collects sample statistics on the acquired
+     * buffers.
+     * @see SampleStatistic
+     */
+    @ManagedObject
+    class SampledPool implements RetainableByteBufferPool, Dumpable
+    {
+        private final RetainableByteBufferPool _pool;
+        private final SampleStatistic _heap = new SampleStatistic();
+        private final SampleStatistic _direct = new SampleStatistic();
+
+        public SampledPool(RetainableByteBufferPool pool)
+        {
+            _pool = pool;
+        }
+
+        @ManagedAttribute("Heap acquire samples")
+        public SampleStatistic getHeapSample()
```

Comment:
Unfortunately `SampleStatistic` is not manageable, so this JMX exposure results in no data being exposed.
I think we need either more specific getters, or MXBean-ify `SampleStatistic` which may not be easy.

I would make this class package private in `ArrayRBBP` for now, and eventually open it up later.

Reply:
If it is private, then we can't use it to gather any info.
Anyway, I'm going to look at sampling on buckets rather than on acquire.